### PR TITLE
[CI] pin max torch version to fix CI errors

### PR DESCRIPTION
**Pull request by a-r-r-o-w**
# What does this PR do?

Only pinned until we figure out what's going on with torch 2.5.0

Context: https://huggingface.slack.com/archives/C065E480NN9/p1729230272869569

#9691, #9704

- https://github.com/huggingface/diffusers/actions/runs/11398910363/job/31721831848?pr=9691
- https://github.com/huggingface/diffusers/actions/runs/11398910363/job/31721832317?pr=9691

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -130,7 +130,7 @@
     "regex!=2019.12.17",
     "requests",
     "tensorboard",
-    "torch>=1.4",
+    "torch>=1.4,<=2.4.1",
```
Comment (sayakpaul) [inline setup.py:133]: Why not < `2.5.1`?

**Comment by sayakpaul** [comment]
Let's also trigger Docker build CI: https://github.com/huggingface/diffusers/actions/workflows/build_docker_images.yml

File: setup.py (hunk 1 above)
Comment (a-r-r-o-w) [inline setup.py:133]:
```suggestion
    "torch>=1.4,<2.5.0",
```

**Comment by a-r-r-o-w** [comment]
Seems like everything passed: https://github.com/huggingface/diffusers/actions/runs/11403577400

**Review by sayakpaul — APPROVED** [review]
LGTM. IIRC you already triggered the CI, so should be good.

**Pull request merged**